daemon/daemonbuilder/builder.go

```diff
@@ -84,16 +92,14 @@ func (d Docker) ContainerUpdateCmd(cID string, cmd []string) error {
 	return nil
 }
 
-// Retain retains an image avoiding it to be removed or overwritten until a corresponding Release() call.
-func (d Docker) Retain(sessionID, imgID string) {
-	// FIXME: This will be solved with tags in client-side builder
-	//d.Daemon.Graph().Retain(sessionID, imgID)
-}
-
-// Release releases a list of images that were retained for the time of a build.
-func (d Docker) Release(sessionID string, activeImages []string) {
-	// FIXME: This will be solved with tags in client-side builder
-	//d.Daemon.Graph().Release(sessionID, activeImages...)
+// ContainerAttach attaches streams to the container cID. If stream is true, it streams the output.
+func (d Docker) ContainerAttach(cID string, stdin io.ReadCloser, stdout, stderr io.Writer, stream bool) error {
+	return d.Daemon.ContainerWsAttachWithLogs(cID, &daemon.ContainerWsAttachWithLogsConfig{
```

Comment:
I feel like we can move this config to `api/types/configs.go`, as well as `ContainerAttachWithLogsConfig`, but that might be better done in a different PR.